discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/Discovery.java

```diff
@@ -10,445 +10,705 @@
+   * @return the {@link ListCollectionsResponse} with the response
+   */
+  public ServiceCall<ListCollectionsResponse> listCollections(ListCollectionsOptions listCollectionsOptions) {
+    Validator.notNull(listCollectionsOptions, "listCollectionsOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.get(String.format("/v1/environments/%s/collections", listCollectionsOptions
+        .environmentId()));
+    builder.query(VERSION, versionDate);
+    if (listCollectionsOptions.name() != null) {
+      builder.query("name", listCollectionsOptions.name());
+    }
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(ListCollectionsResponse.class));
+  }
+
+  /**
+   * Update a collection.
+   *
+   * @param updateCollectionOptions the {@link UpdateCollectionOptions} containing the options for the call
+   * @return the {@link Collection} with the response
+   */
+  public ServiceCall<Collection> updateCollection(UpdateCollectionOptions updateCollectionOptions) {
+    Validator.notNull(updateCollectionOptions, "updateCollectionOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.put(String.format("/v1/environments/%s/collections/%s",
+        updateCollectionOptions.environmentId(), updateCollectionOptions.collectionId()));
+    builder.query(VERSION, versionDate);
+    final JsonObject contentJson = new JsonObject();
+    if (updateCollectionOptions.name() != null) {
+      contentJson.addProperty("name", updateCollectionOptions.name());
+    }
+    if (updateCollectionOptions.description() != null) {
+      contentJson.addProperty("description", updateCollectionOptions.description());
+    }
+    if (updateCollectionOptions.configurationId() != null) {
+      contentJson.addProperty("configuration_id", updateCollectionOptions.configurationId());
+    }
+    builder.bodyJson(contentJson);
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Collection.class));
+  }
+
+  /**
+   * Add configuration.
+   *
+   * Creates a new configuration. If the input configuration contains the `configuration_id`, `created`, or `updated`
+   * properties, then they are ignored and overridden by the system, and an error is not returned so that the overridden
+   * fields do not need to be removed when copying a configuration. The configuration can contain unrecognized JSON
+   * fields. Any such fields are ignored and do not generate an error. This makes it easier to use newer configuration
+   * files with older versions of the API and the service. It also makes it possible for the tooling to add additional
+   * metadata and information to the configuration.
+   *
+   * @param createConfigurationOptions the {@link CreateConfigurationOptions} containing the options for the call
+   * @return the {@link Configuration} with the response
+   */
+  public ServiceCall<Configuration> createConfiguration(CreateConfigurationOptions createConfigurationOptions) {
```

Comment:
`CreateConfigurationOptions` has a `configuration` setter (this took special effort to create), so the developer can supply just an `environment_id` and `configuration` if that's what he/she wants.  The extra setters on `CreateConfigurationOptions` also make it easy to create a new configuration from an existing one with just one property (e.g. name) different.